src/Symfony/Component/Messenger/Stamp/RedeliveryStamp.php

```diff
@@ -27,9 +27,33 @@ final class RedeliveryStamp implements StampInterface
     public function __construct(int $retryCount, string $exceptionMessage = null, FlattenException $flattenException = null, \DateTimeInterface $redeliveredAt = null)
```

Comment:
the arguments itself have not been deprecated and the order needs to be changed ($redeliveredAt before exceptionMessage  and flattenException )

Reply:
As discussed on Slack, I'll create a new PR to:
- Remove the $exceptionMessage parameter
- Remove the $flattenException parameter
- Add checks on amount of parameters given and type-checks for those parameters to trigger the deprecation warnings.